### PR TITLE
Next mac cmake

### PR DESCRIPTION
**Pull request by FunMiles**
This is to add Mac OS X cmake compilation support. Includes fixes from Moved from the discussion on the master branch. Includes the pull request from louis-mclaughlin to fix linux compilation.

### CODE REVIEW

**Comment by seanpaultaylor** [comment]
So this is working on Linux too ? Does anyone know if this is working on Windows too?

**Comment by FunMiles** [comment]
Louis has tested on Linux after his fixes and I have re-tested on Mac after his fixes and it still works. The last commit allows for new projects to also support cmake build (modified template). Louis would need to test that too.

**Comment by louis-mclaughlin** [comment]
This compiles and the samples run successfully on Linux.

However you'll need to update template/template-CMakeLists.txt to include the resource copy macro changes as running a new project (generated with newproject.sh) crashes:

create -- Failed to open file 'res/demo.scene'.
loadInternal -- Failed to load scene file 'res/demo.scene'.
t: /home/louis/Documents/GamePlay-FunMiles/gameplay/src/SceneLoader.cpp:41: gameplay::Scene* gameplay::SceneLoader::loadInternal(const char*): Assertion `0' failed.
Aborted (core dumped)

**Comment by FunMiles** [comment]
My mistake, I left the definition of target directory for the resources to the same value as for Mac on other platforms. I've just pushed a correction. Let me know if it fixes it for Linux.

**Comment by louis-mclaughlin** [comment]
Yep, fixed

File: samples/CMakeLists.txt [hunk 1]
```diff
@@ -1,33 +1,76 @@
 include(BuildHelpers.CMakeLists.txt)
 
-include_directories( 
-    ${CMAKE_SOURCE_DIR}/gameplay/src
-    ${CMAKE_SOURCE_DIR}/external-deps/include
+include_directories(
+        ${CMAKE_SOURCE_DIR}/gameplay/src
```
Comment (seanpaultaylor) [inline samples/CMakeLists.txt]: Tabs need removing.

File: samples/CMakeLists.txt [hunk 2]
```diff
@@ -1,33 +1,76 @@
...
+            )
+ELSE(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    add_definitions(-D__linux__)
 
-IF(ARCH_DIR STREQUAL "x64")
-    link_directories(${CMAKE_SOURCE_DIR}/external-deps/lib/linux/x86_64)
-ELSE()
-    link_directories(${CMAKE_SOURCE_DIR}/external-deps/lib/linux/x86)
-ENDIF(ARCH_DIR STREQUAL "x64")
+    IF(ARCH_DIR STREQUAL "x64")
```
Comment (seanpaultaylor) [inline samples/CMakeLists.txt]: More tabs.

File: samples/racer/CMakeLists.txt [hunk 3]
```diff
@@ -1,30 +1,59 @@
 set(GAME_NAME sample-racer)
 
 set(GAME_SRC
-    src/RacerGame.cpp
-    src/RacerGame.h
-)
+        src/RacerGame.cpp
```
Comment (seanpaultaylor) [inline samples/racer/CMakeLists.txt]: Tabs

File: samples/racer/CMakeLists.txt [hunk 4]
```diff
@@ -1,30 +1,59 @@
...
 
-COPY_RES( ${GAME_NAME} )
-COPY_RES_EXTRA( ${GAME_NAME} ${CMAKE_SOURCE_DIR}/gameplay
-    res/logo_powered_white.png 
-    res/shaders/*
-    res/ui/*
-)
+    COPY_RES( ${GAME_NAME} )
+    COPY_RES_EXTRA( ${GAME_NAME} ${CMAKE_SOURCE_DIR}/gameplay
+            res/logo_powered_white.png
```
Comment (seanpaultaylor) [inline samples/racer/CMakeLists.txt]: Tabs

**Pull request merged**